### PR TITLE
Add silver searcher (ag) to vim

### PR DESCRIPTION
**Pull request by christopher-haueter**
Needed for `:Ag` (10x faster `:grep` replacement) to work in vim

Hit `ctrl k` over any symbol and it will immediately be `ag`ed (e.g. grepped) and the quickfix window will be opened for you to traverse.

### CODE REVIEW

File: vimrc [hunk 1]
```diff
@@ -221,3 +221,5 @@ else
   let &t_EI = "\<Esc>]50;CursorShape=0\x7"
 end
 
+let g:ag_working_path_mode="r"
+nnoremap K :Ag "\b<C-R><C-W>\b"<CR>:cw<CR>
```
Comment (chrishaum) [inline vimrc]: FYI: https://github.com/rking/ag.vim#configuration

File: vimrc [hunk 2]
```diff
@@ -221,3 +221,5 @@ else
   let &t_EI = "\<Esc>]50;CursorShape=0\x7"
 end
 
+let g:ag_working_path_mode="r"
```
Comment (christopher-haueter) [inline vimrc:224]: FYI: https://github.com/rking/ag.vim#configuration

**Comment by christopher-haueter** [comment]
@mwenger1 This is the thing I used while we were pairing, that immediately searches the codebase for the symbol your cursor is over.

**Comment by mwenger1** [comment]
Very excited to use this :ship:

**Comment by arktisklada** [comment]
:rocket:

**Comment by christopher-haueter** [comment]
FYI merged, but wasn't reflected for some reason.

**Pull request closed without merging**